iree/BUILD

```diff
@@ -9,5 +9,3 @@ package(
     features = ["layering_check"],
```

Comment:
Does this (and the accompanying cmake file) need to exist at all?

Reply:
For now, it provides a Bazel package (so we don't skip from `/` straight to `/iree/test/` without a package in `/iree/`). Once `iree/test/` is moved to a new top level directory, this will go away.

Reply:
Yeah but why do we care if we have a Bazel package with nothing in it called IREE? I don't think that does anything

Reply:
I've ran into issues (such as with globs) when one of the intermediate directories is missing a BUILD file. The iree/ directory will be going away soon anyways, so I'd rather not risk it.